### PR TITLE
Add functionality to store user preferences

### LINKED ISSUE #678: [Common] Ability to store user preferences
**Is your feature request related to a problem? Please describe.**
In the current developer portal, we are unable to store the user preferences and therefore they'll be lost when the user logs out of the application.

**Describe the solution you'd like**
The better approach to handle this would be to store these preferences in the local storage so that even if the user logs out the required data will be persisted.

### PR DESCRIPTION
**Pull request by NipuniBhagya**
```
tenantDomain: {
	username: {
		identityAppsSettings: {
		        applicationPreferences: {
		            recentApplications: []
		        },
		        userPreferences: {
		            userListColumns: []
		        }
		}
	}
}
```

### CODE REVIEW

File: apps/admin-portal/src/app.tsx [hunk 1]
```diff
@@ -105,6 +109,34 @@ export const App = (): ReactElement => {
             });
     }, [ config ]);
 
+    /**
+     * Set the application settings of the user to the local storage.
+     */
+    useEffect(() => {
+        if (userName === "") {
```
Comment (brionmario) [inline apps/admin-portal/src/app.tsx:116]: Wouldn't it be better to check all the falsy values .

```suggestion
        if (!userName) {
```

File: apps/admin-portal/src/pages/users.tsx [hunk 2]
```diff
@@ -69,16 +77,29 @@ export const UsersPage: React.FunctionComponent<any> = (): ReactElement => {
...
-            ["meta.created", ""]
-        ]));
+
+        if(CommonHelpers.lookupKey(tenantSettings, username) !== null) {
+            const userSettings = CommonHelpers.lookupKey(tenantSettings, username);
+            const userPreferences = userSettings[1];
+            const tempColumns = new Map<string, string> ();
+
+            if (userPreferences.identityAppsSettings.userPreferences.userListColumns.length < 1) {
+                const metaColumns = ["name", "emails", "userName", "id", "profileUrl", "meta.lastModified"];
```
Comment (brionmario) [inline apps/admin-portal/src/pages/users.tsx:87]: Shall we move these to the constants?

File: modules/core/src/constants/storage-constants.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * under the License.
+ */
+
+/**
+ * Identity apps key in local storage.
+ * @constant
+ * @type {string}
+ * @default
+ */
+export const IDENTITY_APPS_KEY = "identityAppsSettings";
```
Comment (brionmario) [inline modules/core/src/constants/storage-constants.ts:25]: Shall we move these constants to a class?

File: modules/core/src/models/storage.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (brionmario) [inline modules/core/src/models/storage.ts:2]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: modules/core/src/models/storage.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+ */
+
+/**
+ * Interface for the applications settings local storage item.
+ */
+export interface StorageApplicationSettingsInterface {
+    /**
+     * A user's application settings.
+     */
+    [ key: string ]: StorageApplicationSettingsItemsInterface;
```
Comment (brionmario) [inline modules/core/src/models/storage.ts:26]: Can't we pass in this interface using generics?

File: apps/admin-portal/src/app.tsx [hunk 6]
```diff
@@ -45,6 +45,9 @@ import { Helmet } from "react-helmet";
 import { I18n } from "@wso2is/i18n";
 import { I18nextProvider } from "react-i18next";
 import { ThemeContext } from "@wso2is/react-components";
+import { emptyIdentityAppsSettings } from "@wso2is/core/dist/src/models/storage";
```
[the remark this answers is not in the document]
Comment (NipuniBhagya) [reply apps/admin-portal/src/app.tsx:48]: I tried shortening the path but the IDE was unable to resolve the function.

File: apps/admin-portal/src/pages/users.tsx [hunk 7]
```diff
@@ -16,6 +16,9 @@
  * under the License.
  */
 
+import { AuthenticateSessionUtil, AuthenticateUserKeys } from "@wso2is/authentication";
+import { LocalStorageUtils } from "@wso2is/core/dist/src/utils/storage-utils";
```
Comment (thivi) [inline apps/admin-portal/src/pages/users.tsx:20]:
```suggestion
import { LocalStorageUtils } from "@wso2is/core/utils";
```

File: apps/admin-portal/src/pages/users.tsx [hunk 8]
```diff
@@ -16,6 +16,9 @@
...
+import { CommonHelpers } from "@wso2is/core/dist/src/helpers";
```
Comment (thivi) [inline apps/admin-portal/src/pages/users.tsx:21]:
```suggestion
import { CommonHelpers } from "@wso2is/core/helpers";
```

**Pull request merged**